### ISSUE TITLE
In JS, importing JSON should not give an error

### ISSUE THREAD

**Issue by Thaina** [Microsoft/TypeScript#7071]
Make a json file

```
{
    "season": 1
}
```

Code

```
var test = require("test.json");
```

result

![untitled-1](https://cloud.githubusercontent.com/assets/1042507/13026391/9494ca4a-d258-11e5-9a8f-dbd4fd3c4c2e.jpg)

expect

![untitled-2](https://cloud.githubusercontent.com/assets/1042507/13026392/9a969c52-d258-11e5-99a6-c652a2dfd7e2.jpg)

**Comment by kitsonk** [comment]
This is the domain of plugin loaders and there are mechanism already for describing how a plugin loader would work and #6615 will provide a better mechanism for handling this.

Building this into TypeScript directly wouldn't make sense and likely break lots of other implementations, where people are properly using loader plugins.

**Comment by Thaina** [comment]
@kitsonk Sorry but I'm not sure why it about plugin loader while this case is about intellisense (which I think it is provided by salsa)

**Comment by yortus** [comment]
The compiler *could* do this in theory, parsing the JSON structure in a similar way to how it would parse an object literal in a `.ts` or `.js` file. But as @kitsonk points out this is *not* a standardised language-level feature. Node.js allows you to require JSON files, but it's not part of any standard so it's unlikely to be implemented directly in TypeScript.

Having said that, `JSX` parsing *is* implemented directly in TypeScript even though that's not a standard. So it's probably more a question of demand and cost/benefit. If hundreds of +1s start appearing in this issue, who knows? I personally doubt that there's a lot of demand for this, and it's pretty easy to add your own typings for JSON documents anyway.

**Comment by Thaina** [comment]
@yortus I think both `require` and `module` is also not standard js. It is standard of [just for nodejs](https://nodejs.org/api/modules.html) which also standardize load json file as module

I'm not sure how salsa implement, which part call which compiler, however salsa could read all those weird jsdoc and constructor to parse into intellisense so just include require json file should be in the same place as putting other intellisense

**Comment by billti** [comment]
It would be doable with some work, and as commented, we already have some Node specific behavior. That said, as others have commented, this is a very little used feature, so unless we see a lot of demand for it, I don't see it reaching the top of our TODO list any time soon.

**Comment by kitsonk** [comment]
I am not 100% sure in Salsa, but isn't there a way to assert the an object literal type in JSDoc for this, something like:

```js
/**
 * @type {Foo}
 */
var foo = require('foo.json');
```

In TypeScript that is generally what I do when loading JSON and while in theory the structures could get out of sync at design time, they of course can get out of sync at run-time too, so having them "automagically" determined at run-time is of somewhat limited value.

**Comment by RyanCavanaugh** [comment]
Accepting PRs on this if anyone would like to give it a shot. This should only be allowed when the module system is `commonjs`

**Comment by RichiCoder1** [comment]
I'd be interested in taking a swing at this, but I haven't worked with Typescripts internals before. Any advice beyond the general contributing guidance?

**Comment by basarat** [comment]
Minor: that require statement should be most likely have a relative path prefix `require('./test.js')` because of how node modules work :rose:

**Comment by mhegazy** [comment]
I would start by looking at:
https://github.com/Microsoft/TypeScript/wiki/Architectural-Overview and https://github.com/Microsoft/TypeScript/wiki/Contributing-to-TypeScript

I would assume the first step is getting a JSON parser in place. i would expect that to be based on our current parser, possibly with an option to createSourceFile, or just a new ScriptKind.
We will need to then wire module resolution to accept `.json` files, along with other extensions.
Then wire the type system pieces to get the type of a JSON object, that should not be too hard.

**Comment by Thaina** [comment]
Thanks for you all hard work

There is guideline from node people I think it would be useful

https://nodejs.org/api/modules.html#modules_all_together

**Comment by mischkl** [comment]
+1 it just feels natural to be able to do this if you're used to doing it within Node. Making it a built-in feature would ease the transition to TypeScript for Node-based projects.

**Comment by aluanhaddad** [comment]
:+1:

**Comment by piotrwitek** [comment]
+1

**Comment by zheeeng** [comment]
+1 For this feature to benefits usage such as importing JSON configurations/locales to projects.
